test/common/access_log/access_log_impl_test.cc

```diff
@@ -45,12 +46,123 @@ envoy::config::accesslog::v3::AccessLog parseAccessLogFromV3Yaml(const std::stri
   return access_log;
 }
 
+struct AccessLogDestinationTest : public testing::Test {
+  AccessLogDestinationTest() : file_(new MockAccessLogFile()) {}
+  NiceMock<Server::Configuration::MockFactoryContext> context_;
+  NiceMock<Runtime::MockLoader> runtime_;
+  NiceMock<Envoy::AccessLog::MockAccessLogManager> log_manager_;
+  std::shared_ptr<MockAccessLogFile> file_;
+};
+
+TEST_F(AccessLogDestinationTest, File) {
+  const std::string yaml = R"EOF(
+name: accesslog
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
+  path: /dev/null
+  )EOF";
+
+  ON_CALL(context_, runtime()).WillByDefault(ReturnRef(runtime_));
+  ON_CALL(context_, accessLogManager()).WillByDefault(ReturnRef(log_manager_));
+  EXPECT_CALL(log_manager_,
+              createAccessLog(testing::Matcher<const Envoy::Filesystem::FilePathAndType&>(_)))
+      .WillOnce(Invoke(
+          [this](const Envoy::Filesystem::FilePathAndType& file_info) -> AccessLogFileSharedPtr {
+            EXPECT_EQ(file_info.path_, "/dev/null");
+            EXPECT_EQ(file_info.file_type_, Filesystem::DestinationType::File);
+            return file_;
+          }));
+  InstanceSharedPtr log = AccessLogFactory::fromProto(parseAccessLogFromV3Yaml(yaml), context_);
+}
+
+TEST_F(AccessLogDestinationTest, FileThrows) {
```

Comment:
nit: FileWithoutPathThrows

Reply:
fixed in 4b20da95d95619ed84a0c7ee698367acf8d882a2